game/scripts/vscripts/components/courier/courier.lua

```diff
@@ -15,8 +15,25 @@ function Courier:Init ()
 end
 
 function Courier:SpawnCourier (hero)
-  DebugPrint("Creating Courier for Team " .. hero:GetTeamNumber())
-  courier = hero:AddItemByName('item_courier')
-  flying = hero:AddItemByName('item_flying_courier')
-  Courier.hasCourier[hero:GetTeamNumber()] = true
+  Timers:CreateTimer(1, function ()
+    DebugPrint("Creating Courier for Team " .. hero:GetTeamNumber())
+
+    -- Check if there is an item blocking slot 1, if so sell it
+    slot1Item = hero:GetItemInSlot(0)
+    if slot1Item then
+      hero:SellItem(slot1Item)
+    end
+
+    -- Create couriers and then cast them straight away
+    local courier = hero:AddItemByName('item_courier')
+    if courier then
+        hero:CastAbilityImmediately(courier, hero:GetPlayerID())
+    end
+    local flying = hero:AddItemByName('item_flying_courier')
+    if flying then
+        hero:CastAbilityImmediately(flying, hero:GetPlayerID())
+    end
+
```

Comment:
Could we then do `hero:BuyItem(slot1Item)` ? I get that this works, but kinda bad user design to just have items disappear from a players inventory.